sdk/rust-examples/prime-numbers-generator/keep-primes/src/main.rs

```diff
@@ -0,0 +1,72 @@
+//! Prime Numbers Generator : Remove all non-prime numbers
+//!
+//! ## Context
+//!
+//! Remove all non-prime numbers from the number-set
+//!
+//! ## Authors
+//!
+//! The Veracruz Development Team.
+//!
+//! ## Copyright
+//!
+//! See the file `LICENSING.markdown` in the Veracruz root directory for licensing and
+//! copyright information.
+
+use std::fs;
+use anyhow;
+
+const FILENAME: &'static str = "/output/number-set.txt"; 
+
+fn main() -> anyhow::Result<()>{
+
+    let content = String::from_utf8(fs::read(FILENAME)?)?;
+
+    let mut num_vec: Vec<u32> = content
+        .trim()
+        .split(",")
+        .map(|x| x.parse::<u32>().expect("Number couldn't be parsed"))
```

Comment:
Sorry for the delay in reviewing here (we just had a rebuttal period for our paper).  However I think error reporting can be improved here further, rather than aborting at runtime if a line cannot be parsed as a `u32`:

```rust
let mut num_vec = vec![];

for x in content.trim().split(",") {
  if let Some(n) = x.parse::<u32>() {
      num_vec.push(n);
  } else {
      bail!("Unable to parse {} as u32.", x);
  }
}
```

Note that the `bail!()` macro is equivalent to `return Err(Error::msg(...)));` from the `anyhow` crate, and therefore exits the program gracefully with an error code, rather than raising a runtime exception.

Reply:
Done. just a minor detail, `x.parse::<u32>` returns a `Result` not `Option`, so it should be `Ok(n)`.

And thank your for your time and patience on this PR!